src/main/java/org/wso2/carbon/module/swiftiso20022/utils/MT940JSONParserUtils.java

```diff
@@ -0,0 +1,157 @@
+        JsonObject textBlock = mt940JsonObj.getAsJsonObject("textBlock");
+        // Remove NO LETTER OPTIONS
+        textBlock.getAsJsonObject("transactionReferenceNumber").remove("option");
+
+        if (textBlock.has("relatedReference") &&
+                textBlock.getAsJsonObject("relatedReference").get("option").getAsCharacter() ==
+                        MTParserConstants.FIELD_OPTION_NO_LETTER) {
+            textBlock.getAsJsonObject("relatedReference").remove("option");
+        }
+
+        if (textBlock.getAsJsonObject("accountIdentification").get("option").getAsCharacter() ==
+                MTParserConstants.FIELD_OPTION_NO_LETTER) {
+            textBlock.getAsJsonObject("accountIdentification").remove("option");
+        }
+
+        if (textBlock.has("statementLines")) {
+            for (JsonElement statement : textBlock.getAsJsonArray("statementLines")) {
+                if (statement.getAsJsonObject().has("Field86") && statement.getAsJsonObject()
+                        .getAsJsonObject("Field86").get("option").getAsCharacter() ==
+                        MTParserConstants.FIELD_OPTION_NO_LETTER) {
+
+                    statement.getAsJsonObject().getAsJsonObject("Field86").remove("option");
+                }
+
+                // Update the Field61 amount
+                Double transactionAmt = MTParserUtils.convertMTAmountToISOAmount(
+                        statement.getAsJsonObject().getAsJsonObject("Field61").get("amount").getAsString());
+
+                statement.getAsJsonObject().getAsJsonObject("Field61").remove("amount");
+                statement.getAsJsonObject().getAsJsonObject("Field61").addProperty("amount", transactionAmt);
+            }
+        }
+
+        if (textBlock.has("infoToAccountOwner") &&
+                textBlock.getAsJsonObject("infoToAccountOwner").get("option").getAsCharacter() ==
+                MTParserConstants.FIELD_OPTION_NO_LETTER) {
+            textBlock.getAsJsonObject("infoToAccountOwner").remove("option");
+        }
+
+        // Update the balance amounts
+        String[] balanceTypes = new String[] { "openingBal", "closingBal", "closingAvlBalance", "forwardAvlBalance" };
+        for (String balanceType : balanceTypes) {
+            if (!textBlock.has(balanceType)) {
+                continue;
+            }
+
+            Double balanceAmt = MTParserUtils.convertMTAmountToISOAmount(
+                    textBlock.getAsJsonObject(balanceType).get("amount").getAsString());
+
+            textBlock.getAsJsonObject(balanceType).remove("amount");
+            textBlock.getAsJsonObject(balanceType).addProperty("amount", balanceAmt);
+        }
+    }
+
+    /**
+     * Update the dates in the MT940 for processing freemarker template to MT940 to ISO convert.
+     * @param mt940JsonObj         JSON object tree for MT940 message
+     */
+    public static void updateDatesFrMT940(JsonObject mt940JsonObj) {
```

Comment:
Check for spellings

Reply:
Fixed in e2eba5896d2549c41ebd26848c9e0af9148a62ce